libnmstate/varlink/__init__.py

```diff
@@ -0,0 +1,257 @@
+    service = service
+
+
+@service.interface("io.nmstate")
+class NmstateService:
+    def Show(self, include_status_data):
+        """
+        Reports the state data on the system
+        """
+        show_kwargs = {}
+        if include_status_data is not None:
+            show_kwargs["include_status_data"] = include_status_data
+        try:
+            configured_state = libnmstate.show(**show_kwargs)
+            state = {"state": configured_state}
+            return state
+        except libnmstate.error.NmstateValueError as exception:
+            raise NmstateValueError(str(exception))
+
+    def Apply(self, state, verify_change, commit, rollback_timeout):
+        """
+        Apply desired state declared in json format
+        """
+        if state is None:
+            raise varlink.InvalidParameter("state")
+        apply_kwargs = {}
+        if verify_change is not None:
+            apply_kwargs["verify_change"] = verify_change
+        if commit is not None:
+            apply_kwargs["commit"] = commit
+        if rollback_timeout is not None:
+            apply_kwargs["rollback_timeout"] = rollback_timeout
+        try:
+            libnmstate.apply(state, **apply_kwargs)
+        except TypeError:
+            raise varlink.InvalidParameter
+        except libnmstate.error.NmstatePermissionError as exception:
+            raise NmstatePermissionError(str(exception))
+        except libnmstate.error.NmstateConflictError as exception:
+            raise NmstateConflictError(str(exception))
+        except libnmstate.error.NmstateVerificationError as exception:
+            raise NmstateVerificationError(str(exception))
+
+    def Commit(self, checkpoint):
+        """
+        Commits the checkpoint
+        """
+        commit_kwargs = {}
+        if checkpoint is not None:
+            commit_kwargs["checkpoint"] = checkpoint
+        try:
+            libnmstate.commit(**commit_kwargs)
+        except libnmstate.error.NmstateValueError as exception:
+            raise NmstateValueError(str(exception))
+
+    def Rollback(self, checkpoint):
+        """
+        Roll back to the checkpoint
+        """
+        rollback_kwargs = {}
+        if checkpoint is not None:
+            rollback_kwargs["checkpoint"] = checkpoint
+        try:
+            libnmstate.rollback(**rollback_kwargs)
+        except libnmstate.error.NmstateValueError as exception:
+            raise NmstateValueError(str(exception))
+
+
+def get_varlink_server(address):
+    """
+    Returns varlink server object
+    Checks if the varlink address already in use.
+    """
+    server = varlink.ThreadingServer(
+        address, ServiceRequestHandler, bind_and_activate=False
+    )
+    server.allow_reuse_address = True
+    try:
+        server.server_bind()
+        server.server_activate()
+    except OSError as exception:
+        server.server_close()
+        if exception.errno == OS_ERROR_98_CODE:
```

Comment:
Please try to name the constant after its purpose and not its content, so something like `OS_ERROR_ADDRESS_ALREADY_IN_USE` would make it more clear what is checked, here.